### DISCUSSION TITLE
HomeAssistant wont connect to Ginlong or Soliscloud

### DISCUSSION THREAD

**Discussion by ceemjay** [hultenvp/solis-sensor#283] [General]
Tried all combinations but cannot get HA to connect - just keep getting "Cannot login with provided URL and credentials"

I assume Ginlong Plant ID and Soliscloud Station ID are the same 19 digit number? There is a Plant ID on the soliscloud website but it ends with a capital B and that digit cannot be entered.

Are there any error logs I could inspect?

Any help would be greatly appreciated. Thanks

**Comment by himBabbar** [comment]
The soliscloud_api.py   class SoliscloudConfig(PortalConfig): doesn't return the plant_id as a result the API call returns error and doesn't find the inverter. 

following change fixes the issue ;
     add this line to the class definition in soliscloud_api.py
        self._plant_id: str = portal_plantid

 @property
    def plant_id(self) -> str:
        """ Key ID."""
        return self._plant_id

once this was added, it worked like a charm

**Comment by ceemjay** [comment]
@Property <https://github.com/Property>  Thanks for the speedy reply.

I am new to HA and not overly familiar with Python but I have made the
changes and it still doesn't work.

I have a couple of questions:

1. Why has this not been corrected in the latest version on Github (you may
not know the answer but hopefully someone else does)

2. Are these changes required for both Ginlong2 and Solisclound

3..In your code, is """ Key ID.""" correct? Shouldn't it be the plant ID?

4, I assume @Property <https://github.com/Property> was a type and it
should be @property (the original threw an error)?

My code follows with my changes in RED is it correct?

class SoliscloudConfig(PortalConfig):

    """ Portal configuration data """



    def __init__(self,

        portal_domain: str,

        portal_username: str,

        portal_key_id: str,

        portal_secret: bytes,

        portal_plantid: str

    ) -> None:

        super().__init__(portal_domain, portal_username, portal_plantid)

        self._key_id: str = portal_key_id

        self._plant_id: str = portal_plantid

        self._secret: bytes = portal_secret

        self._workarounds = {}

        try:

            with open('/config/custom_components/solis/workarounds.yaml',
'r') as file:

                self._workarounds = yaml.safe_load(file)

                _LOGGER.debug("workarounds: %s", self._workarounds)

        except FileNotFoundError:

            pass



    @property

    def key_id(self) -> str:

        """ Key ID."""

        return self._key_id



    @property

    def secret(self) -> bytes:

        """ API Key."""

        return self._secret



    @property

    def workarounds(self) -> dict[str, Any]:

        """ Return all workaround settings """

        return self._workarounds

**Reply by himBabbar** [reply]
sorry was busy with something so couldn't respond. 

yes it should be @property. 

I also spotted that the API name is also not correct. As per solis documentation the API name is - INVERTER_DETAIL = '/v1/api/inveterDetail'. if you see the 'r' is missing from inverter. this is on line 48 of soliscloud_api.py. make this change and it will work. worked for me. I missed to add this in my original recommendation.

**Reply by hultenvp** [reply]
Thanks @himBabbar,

Regarding the inveterDetail: the first versions of the API had this typo in the endpoint. They now have corrected that and support both for compatibility reasons. The latest versions of the integration already use the endpoint with the correct naming.

**Comment by brewmark27** [comment]
I have the same issue. I've tried to add the lines off code like you mentioned, but this doesn't solve the problem for me.
Are there any other suggestions to fix this?

**Comment by ceemjay** [comment]
Sorry but I couldnt get it to work so gave up and used a different add-on

**Comment by nao982** [comment]
05 Dec 2024
Solis integration station is not loaded
Logger: homeassistant.setup
Source: setup.py:269
First occurred: 12:21:43 PM (1 occurrences)
Last logged: 12:21:43 PM

Setup failed for custom integration 'solis': Requirements for solis not found: ['aiofiles>=23.1.0,<24.0.0'].

**Reply by nao982** [reply]
Please help to advise, thanks

**Comment by ambardesign** [comment]
I was able to add the inverter. The trick is to use the numeric ID from the url https://www.soliscloud.com/#/station/stationDetails/generalSituation/[this_ID] and not the key  displayed on the dashboard.
![2025-04-05_101200](https://github.com/user-attachments/assets/4f93c80e-628c-4263-9f1f-dbe39e937856)

**Reply by nao982** [reply]
thank you

**Reply by nao982** [reply]
since the refreshment data will happen in every 5 minutes from soliscloud, it is not good to use signals for automation in home assistant.
Could you advice is there any other way to add solis inverter into home assistant with getting real time signal updating which may be every 5-10 seconds.

Noted: I'm using Solis Hybrid S6-EH1P6-L-Plus inverter

**Reply by ambardesign** [reply]
At my previous PV plant I used the MQTT-integration to make this work. Right now, on this location, I don't need this setup.
